src/drivers/net/e1000.c

```diff
@@ -282,18 +297,18 @@ static int e1000_open(struct net_device *dev) {
 	mdelay(MDELAY);
 	REG_ORIN(e1000_reg(dev, E1000_REG_RCTL),  E1000_REG_RCTL_MPE);
 
-	for (size_t i = 0; i < E1000_RXDESC_NR; i ++) {
-	        struct sk_buff *skb;
-		skb = skb_alloc(ETH_FRAME_LEN);
-		if (skb == NULL)
-			panic("Can't allocate skb %d", i);
-		rx_skbs[i]= skb;
-		rx_descs[i].buffer_address = (uint32_t) skb->mac.raw;
+	for (int i = 0; i < E1000_RXDESC_NR; i ++) {
+	        struct sk_buff *skb = skb_alloc(E1000_MAX_RX_LEN);
+		if (skb == NULL) {
+			e1000_stop(dev);
+			return -ENOMEM;
+		}
+		nic_priv->rx_skbs[i] = skb;
+		nic_priv->rx_descs[i].buffer_address = (uint32_t) skb->mac.raw;
 	}
```

Comment:
Card should be stopped before skb's could be freed.
What you suggest should only result in e1000_open change, e1000_close may be not called from e1000_open, but it will stay same